src/main/java/io/confluent/connect/hdfs/avro/AvroRecordWriterProvider.java

```diff
@@ -52,7 +53,8 @@ public String getExtension() {
       final String filename
   ) {
     return new io.confluent.connect.storage.format.RecordWriter() {
-      final DataFileWriter<Object> writer = new DataFileWriter<>(new GenericDatumWriter<>());
+      final DataFileWriter<Object> writer = new DataFileWriter<>(new GenericDatumWriter<>())
+          .setCodec(CodecFactory.fromString(conf.getAvroCodec()));
```

Comment:
What happens when `conf.getAvroCodec()` returns null?

Reply:
If conf.getAvroCodec() returns a "null" (a String), then CodecFactory returns a No-op codec. This is the default codec used by Avro. If the function returns a null, or an unknown codec (lz4, for example), we get an AvroRuntimeException.

Reply:
This can't happen given that parsing of String values is required by `AbstractConfig`. That's why the default can't be set to `null` and as in the S3 it's instead set to `"null"`.